test/dummy/config/application.rb

```diff
@@ -13,6 +12,7 @@ class Application < Rails::Application
     # -- all .rb files in that directory are automatically loaded.
 
     config.active_job.queue_adapter = :test
-    config.action_mailer.default_url_options = { host: "localhost", port: 3000 }
+    config.action_mailer.default_url_options = {host: "localhost", port: 3000}
+    config.active_record.legacy_connection_handling = false
```

Comment:
#### From running against Rails master:
> ArgumentError: The `legacy_connection_handling` setter was deprecated in 7.0 and removed in 7.1, but is still defined in your configuration. Please remove this call as it no longer has any effect."

#### From running against Rails version 6:
> NoMethodError: undefined method `legacy_connection_handling=' for ActiveRecord::Base:Class

This new internal API for `config.active_record.legacy_connection_handling` came in Rails 6.1+, was deprecated in 7.0 and removed in 7.1

Knowing this, maybe we can do something like:
```ruby
config.active_record.legacy_connection_handling = false if ("6.1".."7.0.4").cover?(Rails.version)
```

Reply:
> Looks like we've got more work to do... sweat_smilesweat_smilesweat_smile

Yes, indeed. I'm up!

Reply:
Count me in :eyes:

Reply:
hey guys, I ended up committing since the code @cjilbert504 sent seemed to work fine. If for some reason the tests still fail due to some other issues, for sure I'm up for getting together again to figure it out :smile:

Reply:
Looks like it worked!